csu

```diff
@@ -13,411 +13,294 @@
+  echo "Running test suite backwards..."
+  if [ "$TRAVIS_PULL_REQUEST" != "false" ]; then
     docker-compose exec django /docker_venv/bin/python3 ./manage.py test --settings=config.settings.testing --pattern "test_*.py" --reverse -v 0
   fi
 }
-function test_suite_backwards_help {
-  printf "   test_backwards    Run test suite backwards.\n"
-}
+defhelp -dev test_suite_backwards 'Run test suite backwards.'
 
 # Generates the documentation (with warnings as errors)
-function documentation {
-  printf "Removing any existing documentation...\n"
+dev_docs() {
+  echo "Removing any existing documentation..."
   docker-compose exec django rm -rf /cs-unplugged/docs/build/
   docker-compose exec django mkdir /cs-unplugged/docs/build/
-  printf "\nCreating documentation...\n"
+  echo
+  echo "Creating documentation..."
   docker-compose exec django /docker_venv/bin/sphinx-build -W /cs-unplugged/docs/source/ /cs-unplugged/docs/build/
 }
-function documentation_help {
-  printf "   docs              Generate documentation.\n"
-}
+defhelp -dev docs 'Generate documentation.'
 
 # Delete all untagged dangling Docker images
-function clean {
-  printf "If the following commands return an argument not found error,\n"
-  printf "this is because there is nothing to delete for clean up.\n"
+cmd_clean() {
+  echo "If the following commands return an argument not found error,"
+  echo "this is because there is nothing to delete for clean up."
 
-  printf "\nDeleting unused volumes...\n"
+  echo
+  echo "Deleting unused volumes..."
   docker volume ls -qf dangling=true | xargs -r docker volume rm
-  printf "\nDeleting exited containers...\n"
+  echo
+  echo "Deleting exited containers..."
   docker ps --filter status=dead --filter status=exited -aq | xargs docker rm -v
-  printf "\nDeleting dangling images...\n"
-  docker rmi $(docker images -f "dangling=true" -q)
-}
-function clean_help {
-  printf "   clean             Delete unused Docker files.\n"
+  echo
+  echo "Deleting dangling images..."
+  docker images -f "dangling=true" -q | xargs docker rmi
 }
+defhelp clean 'Delete unused Docker files.'
 
 # Delete all Docker containers and images
-function wipe {
-  docker rm $(docker ps -a -q)
-  docker rmi $(docker images -q)
-}
-function wipe_help {
-  printf "   wipe              Delete all Docker containers and images.\n"
+cmd_wipe() {
+  docker ps -a -q | xargs docker rm
+  docker images -q | xargs docker rmi
 }
+defhelp wipe 'Delete all Docker containers and images.'
 
 # View logs
-function logs {
+cmd_logs() {
   docker-compose logs
 }
-function logs_help {
-  printf "   logs              View logs.\n"
+defhelp logs 'View logs.'
+
+silent() {
+  "$@" > /dev/null 2>&1
 }
 
-function process_dev_command() {
-  if [ $# -lt 1 ]
-  then
-    printf "${RED}ERROR: dev command requires one parameter!${NC}\n"
-    helpmenu
-    exit 1
+cmd_dev() {
+  local cmd="$1"
+  shift
+  if [ -z "$cmd" ]; then
+    echo -e "${RED}ERROR: dev command requires one parameter!${NC}"
```

Comment:
Answering my own question here, yes `-e` is necessary to match the `\033` escape character for the colour change.